docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,291 @@
+
+<p align="center">
+  <img src="../../examples/platform/nrfconnect/doc/images/Logo_RGB_H-small.png" alt="Nordic Semiconductor logo"/>
+  <img src="../../examples/platform/nrfconnect/doc/images/nRF52840-DK-small.png" alt="nRF52840 DK">
+</p>
+
+<hr>
+This document is based on [Matter](https://github.com/project-chip/connectedhomeip), and Nordic
+Semiconductor's nRF Connect SDK, and describes the process of creating and programming factory data.
+
+-   [Overview](#overview)
+    -   [Factory data components](#factory-data-components)
+    -   [Factory data format](#factory-data-format)
+-   [Using own factory data](#using-own-factory-data)
+-   [Generating factory data](#generating-factory-data)
+    -   [Creating factory data JSON file](#creating-factory-data-json-file)
+    -   [Verifying using a JSON schema](#verifying-using-a-json-schema)
+    -   [Preparing factory data partition on a device](#preparing-factory-data-partition-on-a-device)
+    -   [Generating factory data partition](#generating-factory-data-partition)
+-   [Building an example with factory data](#building-an-example-with-factory-data)
+    -   [Providing factory data parameters as a build argument list](#providing-factory-data-parameters-as-a-build-argument-list)
+    - [Setting factory data parameters using interactive Kconfig interfaces](#setting-factory-data-parameters-using-interactive-kconfig-interfaces)
+-   [Programming factory data](#programming-factory-data)
+
+<hr>
+
+<a name="overview"></a>
+
+## Overview
+
+The factory data parameter set includes information such as a manufacturer name, date of manufacturing, product certifications, and PCB versioning. 
+
+For the nRF Connect platform, factory data is stored by default in the internal Flash memory in a separate partition. This helps to keep factory partition secure by applying hardware protection. Parameters are read at the boot time of a device, and then they can be used in the Matter stack and user application.
+
+> Note: Encryption and security of the factory data partition is not provided yet for this feature.
+
+### Factory data components
+
+A factory data set consists of following parameters:
+
+|Key name|Full name|Length|Format|Conformance|Description|
+|:------:|:-------:|:----:|:----:|:---------:|:---------:|
+|**`sn`**|`serial number`|<1, 32> B|*ASCII string*|mandatory|A serial number parameter defines an unique number of manufactured device. Maximum length of serial number is 32 characters.|
+|**`vendor_id`**|`vendor ID`|2 B|*uint16*|mandatory|A MATTER-assigned id for the organization responsible for producing the device.|
+|**`product_id`**|`product ID`|2 B|*uint16*|mandatory|A unique id assigned by the device vendor to identify the product or device type. Defaults to a MATTER-assigned id designating a non-production or test "product".|
```

Comment:
```suggestion
|**`product_id`**|`product ID`|2 B|*uint16*|mandatory|A unique id assigned by the device vendor to identify the product. Defaults to a MATTER-assigned id designating a non-production or test "product".|
```
"Device type" has a special meaning in the context of Matter so better not to confuse those two things.